dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicSimulation.java

```diff
@@ -32,43 +32,70 @@ public Runner(DynamicSimulationProvider provider) {
+        }
+
+        public CompletableFuture<DynamicSimulationResult> runAsync(Network network, CurvesSupplier curvesSupplier, ComputationManager computationManager,
+                                                                   DynamicSimulationParameters parameters) {
+            return runAsync(network, curvesSupplier, network.getVariantManager().getWorkingVariantId(), computationManager, parameters);
+        }
+
+        public CompletableFuture<DynamicSimulationResult> runAsync(Network network, ComputationManager computationManager, DynamicSimulationParameters parameters) {
             return runAsync(network, network.getVariantManager().getWorkingVariantId(), computationManager, parameters);
         }
 
-        public CompletableFuture<DynamicSimulationResult> runAsync(Network network,
-            DynamicSimulationParameters parameters) {
+        public CompletableFuture<DynamicSimulationResult> runAsync(Network network, CurvesSupplier curvesSupplier, DynamicSimulationParameters parameters) {
+            return runAsync(network, curvesSupplier, LocalComputationManager.getDefault(), parameters);
+        }
+
+        public CompletableFuture<DynamicSimulationResult> runAsync(Network network, DynamicSimulationParameters parameters) {
             return runAsync(network, LocalComputationManager.getDefault(), parameters);
         }
 
+        public CompletableFuture<DynamicSimulationResult> runAsync(Network network, CurvesSupplier curvesSupplier) {
+            return runAsync(network, curvesSupplier, DynamicSimulationParameters.load());
+        }
+
         public CompletableFuture<DynamicSimulationResult> runAsync(Network network) {
             return runAsync(network, DynamicSimulationParameters.load());
         }
 
-        public DynamicSimulationResult run(Network network, String workingStateId,
-            ComputationManager computationManager, DynamicSimulationParameters parameters) {
-            Objects.requireNonNull(workingStateId);
-            Objects.requireNonNull(parameters);
-            return provider.run(network, computationManager, workingStateId, parameters).join();
+        public DynamicSimulationResult run(Network network, CurvesSupplier curvesSupplier, String workingVariantId, ComputationManager computationManager,
+                                           DynamicSimulationParameters parameters) {
+            return runAsync(network, curvesSupplier, workingVariantId, computationManager, parameters).join();
         }
 
-        public DynamicSimulationResult run(Network network, ComputationManager computationManager,
-            DynamicSimulationParameters parameters) {
+        public DynamicSimulationResult run(Network network, String workingVariantId, ComputationManager computationManager, DynamicSimulationParameters parameters) {
+            return runAsync(network, workingVariantId, computationManager, parameters).join();
```

Comment:
Here shouldn't we return run(network, CurvesSuplier.empty(), workingVariantId, computationManager, parameters) instead?